File: primes/primes.rb

div_array = []

(2..ARGV[0].to_i).each do |i|
	s = Math.sqrt(i)
	isprime = true
	for j in div_array 
		break if j > s
		if i % j == 0
			isprime = false
			break
		end
	end
	if isprime
		div_array << i
		p i
	end
end

File: php/euler/7/primes.php
<?
	$n = 1;
	for($i=2;$n<=10001;$i++) {
		if(is_prime($i)) $n++;
	}
	print $i-1 . "\n";
	function is_prime($num) {
		for($i=2;$i<=sqrt($num);$i++) {
			if($num % $i == 0) return false;
		}
		return true;
	}
?>
